sdk/resource/detect/env.go

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package detect
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"os"
+	"strings"
+
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+// Environment variable used by "env" to decode a resource.
+const envVar = "OTEL_RESOURCE_LABELS"
+
+var (
+	//ErrMissingPairValue happens when a pair does not include value
+	ErrMissingPairValue = errors.New("missing pair value")
+
+	//ErrUnescape happens when '%' is not followed by two hexadecimal digits
+	ErrUnescape = errors.New("invalid resource format in attribute")
+)
+
+// FromEnv is a detector that implments the ResourceDetector and collects resources
+// from environment
+type FromEnv struct{}
```

Comment:
A compile time assertion this implements the detector would prevent runtime panics if future errors are introduced.

```suggestion
type FromEnv struct{}

var _ Detector = (*FromEnv)(nil)
```

(note that ^ is not valid based on the current package structure)